include/mesh/mesh_function.h

```diff
@@ -146,27 +146,45 @@ class MeshFunction : public FunctionBase<Number>,
 
   /**
    * Computes values at coordinate \p p and for time \p time, which
-   * defaults to zero.
+   * defaults to zero, optionally restricting the point to the passed
+   * subdomain_ids. This is useful in cases where there are multiple
+   * dimensioned elements, for example.
    */
   void operator() (const Point& p,
                    const Real time,
                    DenseVector<Number>& output);
 
   /**
+   * Computes values at coordinate \p p and for time \p time,
+   * restricting the point to the passed subdomain_ids. This is useful in
+   * cases where there are multiple dimensioned elements, for example.
+   */
+  void operator() (const Point& p,
+                   const Real time,
+                   DenseVector<Number>& output,
+                   const std::set<subdomain_id_type>* subdomain_ids);
+
+  /**
    * Computes gradients at coordinate \p p and for time \p time, which
```

Comment:
The FunctionBase changes can go in a separate PR.  Let's merge this now.

I'm ambivalent about whether moving to one method is worth the hassle.  There's nothing *wrong* about the original API, it's just less flexible.  If you feel strongly (or if someone else does and wants to pipe up) then we could deprecate it, though.